pkg/verifier/verifier.go

```diff
@@ -96,8 +138,17 @@ func (c *cachedVerificationResponses) setCachedResponse(cluster string, gvk sche
 	verifiedResources[gvk] = registered
 }
 
+func (c *cachedVerificationResponses) clearedCachedResponses() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.verifiedClusterResources = map[string]map[schema.GroupVersionKind]bool{}
+}
+
+// NewVerifier returns a new verifier
+// Deprecated: use VerifierFactory instead
 func NewVerifier(
 	ctx context.Context,
+	discoveryClient discovery.DiscoveryInterface,
```

Comment:
this is a breaking change. is that intentional so you can fix all the invocations?